iox_query/src/exec/gapfill.rs

```diff
@@ -604,4 +614,244 @@ mod test {
+            "+----+--------------------------+----+",
+            "| a  | 1970-01-01T00:00:00.975Z |    |",
+            "| a  | 1970-01-01T00:00:01Z     | 10 |",
+            "| a  | 1970-01-01T00:00:01.025Z |    |",
+            "| a  | 1970-01-01T00:00:01.050Z |    |",
+            "| a  | 1970-01-01T00:00:01.075Z |    |",
+            "| a  | 1970-01-01T00:00:01.100Z | 11 |",
+            "| a  | 1970-01-01T00:00:01.125Z |    |",
+            "+----+--------------------------+----+",
+        ];
+        assert_batches_eq!(expected, &batches);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_gapfill_multi_group() -> Result<()> {
+        test_helpers::maybe_start_logging();
+        let batch = TestBatch {
+            group_cols: vec![vec![Some("a"), Some("a"), Some("b"), Some("b")]],
+            time_col: vec![Some(1_000), Some(1_100), Some(1_025), Some(1_050)],
+            agg_cols: vec![vec![Some(10), Some(11), Some(20), Some(21)]],
+        };
+        let params = get_params_ms(&batch, 25, 975, 1_125);
+        let tc = TestCase {
+            in_batches: vec![batch],
+            batch_size: 1000,
+            params,
+        };
+        let batches = tc.run().await?;
+        let expected = [
+            "+----+--------------------------+----+",
+            "| g0 | t                        | a0 |",
+            "+----+--------------------------+----+",
+            "| a  | 1970-01-01T00:00:00.975Z |    |",
+            "| a  | 1970-01-01T00:00:01Z     | 10 |",
+            "| a  | 1970-01-01T00:00:01.025Z |    |",
+            "| a  | 1970-01-01T00:00:01.050Z |    |",
+            "| a  | 1970-01-01T00:00:01.075Z |    |",
+            "| a  | 1970-01-01T00:00:01.100Z | 11 |",
+            "| a  | 1970-01-01T00:00:01.125Z |    |",
+            "| b  | 1970-01-01T00:00:00.975Z |    |",
+            "| b  | 1970-01-01T00:00:01Z     |    |",
+            "| b  | 1970-01-01T00:00:01.025Z | 20 |",
+            "| b  | 1970-01-01T00:00:01.050Z | 21 |",
+            "| b  | 1970-01-01T00:00:01.075Z |    |",
+            "| b  | 1970-01-01T00:00:01.100Z |    |",
+            "| b  | 1970-01-01T00:00:01.125Z |    |",
+            "+----+--------------------------+----+",
+        ];
+        assert_batches_eq!(expected, &batches);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_gapfill_two_output_batches() -> Result<()> {
+        test_helpers::maybe_start_logging();
+        let batch = TestBatch {
+            group_cols: vec![vec![Some("a"), Some("a")]],
+            time_col: vec![Some(1_000), Some(1_100)],
+            agg_cols: vec![vec![Some(10), Some(11)]],
+        };
+        let params = get_params_ms(&batch, 25, 975, 1_125);
+        let tc = TestCase {
+            in_batches: vec![batch],
+            batch_size: 4,
+            params,
+        };
+        let result = tc.run().await;
+        assert!(result.is_err()); // not implemented yet
```

Comment:
If you want to test the actual error message, you can do so with something like

```suggestion
        let err = result.unwrap_err();
        assert_eq!(err.to_string(), "this is the actual message"); 
```

Reply:
I had tried to do it it with `assert_error!` but the error was wrapped up in external and context layers, and it wasn't obvious how to get it out. As it is this test case is passing on my local branch.